### ISSUE TITLE
Use gated checkin to protect trimming wins by linking representative apps and measuring size on disk

### ISSUE THREAD

**Issue by danmoseley** [dotnet/runtime#943]
We want to make CoreFX more trimmable (so customers can trim their apps, and also so that Xamarin can use more of it on devices) and we want to protect any size improvements we make this way.

The idea is to in gated checkin - 
1. Pull down down some representative apps in binary form (perhaps from corefx-testdata repo) along with presumably any SDK they require (such as latest ASP.NET implementation assemblies)
2. Use linker to trim them against current CoreFX implementation AND against previous CoreFX implementation (the latter presumably gotten from recent SDK)
3. Block checkin if size has increased more than a non trivial amount (this is the tricky part, we probably don't want to block adding new code - that should be won back in a separate effort - but we do want to block new interdependencies which may have been inadvertently created)

We may not have such representative apps today, but we can start with Hello World while we figure out what they should be and create/obtain them.

cc @marek-safar @ericstj

**Comment by ericstj** [comment]
@joperezr I think you did this.

**Comment by joperezr** [comment]
not really, we logged https://github.com/dotnet/runtime/issues/37258 which was the issue that was going to track the work of adding infrastructure for tests to make sure annotations added to the libraries would root dynamic dependencies (or basically tests checking correctness of the annotations). 

This other issue is very similar but is intended more to track size gains/regressions instead of correctness. The idea for this is to add some tests in the performance repo that will run in some cadence and ensure there are no big size regressions on each run.

**Comment by eerhardt** [comment]
Moving to 6.0 as this won't be completed in 5.0.
